bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java

```diff
@@ -198,29 +199,31 @@ public void processMessage(String topic, byte[] payload) {
             return;
         }
 
-        Command parsedCommand;
+        Type parsedType;
         // Map the string to a command, update the cached value and post the command to the framework
         try {
-            parsedCommand = cachedValue.parseMessage(command);
+            parsedType = cachedValue.parseMessage(command);
         } catch (IllegalArgumentException | IllegalStateException e) {
             logger.warn("Command '{}' from channel '{}' not supported by type '{}': {}", strValue, channelUID,
                     cachedValue.getClass().getSimpleName(), e.getMessage());
             receivedOrTimeout();
             return;
         }
 
-        // things that are only Commands _must_ be posted as a command (like STOP)
-        if (!(parsedCommand instanceof State)) {
-            channelStateUpdateListener.postChannelCommand(channelUID, parsedCommand);
+        if (parsedType instanceof State parsedState) {
+            cachedValue.update(parsedState);
+        } else {
+            // things that are only Commands _must_ be posted as a command (like STOP)
+            channelStateUpdateListener.postChannelCommand(channelUID, (Command) parsedType);
             receivedOrTimeout();
             return;
         }
-        cachedValue.update((State) parsedCommand);
 
-        if (config.postCommand) {
-            channelStateUpdateListener.postChannelCommand(channelUID, (Command) cachedValue.getChannelState());
+        State newState = cachedValue.getChannelState();
+        if (config.postCommand && newState instanceof Command newCommand) {
```

Comment:
`/**
     * If true, the channel state is not updated on a new message.
     * Instead a postCommand() call is performed.
     */
    public boolean postCommand = false;`

`<description>If the received MQTT value should not only update the state of linked items, but command them, enable
				this option.</description>`

By this contract the channel state should not be updated when postCommand is true.

UPDATE: probably field's commentary should be updated.